evalml/tests/data_checks_tests/test_datetime_nan_data_check.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import pandas as pd
+import woodwork as ww
+
+from evalml.data_checks import (
+    DataCheckError,
+    DataCheckMessageCode,
+    DateTimeNaNDataCheck
+)
+
+
+def test_highly_null_data_check_errors(ts_data):
+    data, _ = ts_data
+    data.reset_index(inplace=True, drop=False)
+    data.at[0, 'index'] = np.NaN
+    dt_nan_check = DateTimeNaNDataCheck()
+    assert dt_nan_check.validate(data) == {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (index) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={"column": 'index'})]
+    }
+
+
+def test_highly_null_data_check_input_formats():
+    dt_nan_check = DateTimeNaNDataCheck()
+
+    # test empty pd.DataFrame
+    assert dt_nan_check.validate(pd.DataFrame()) == {"warnings": [], "errors": [], "actions": []}
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (index) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={"column": 'index'})]
+    }
+
+    dates = np.arange(np.datetime64('2017-01-01'), np.datetime64('2017-01-08'))
+    dates[0] = np.datetime64('NaT')
+
+    #  test Woodwork
+    ww_input = ww.DataTable(pd.DataFrame(dates, columns=['index']))
+    assert dt_nan_check.validate(ww_input) == expected
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (0) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={'column': 0})]
+    }
+
+    #  test 2D list
```

Comment:
Typo: this is a 1d list right?

Reply:
theres two columns! `dates` and `np.arange(..)`